scripts/tests/AUTH/test_3b_key_migration_execute.py

```
@@ -0,0 +1,584 @@
+    validates that only a single SELECT statement is allowed.
+
+    Raises:
+        ValueError: If SQL is not a single SELECT statement
+        RuntimeError: If sqlite3 command fails
+    """
+    # Normalize: strip whitespace and trailing semicolons
+    sql_normalized = sql.strip().rstrip(";").strip()
+
+    # Reject multi-statement input (internal semicolons)
+    if ";" in sql_normalized:
+        raise ValueError(f"run_docker_sql only allows single statements, got: {sql[:50]}")
```

Ruff (0.14.14)

[warning] Avoid specifying long messages outside the exception class

(TRY003)